lib/https.c

```diff
@@ -349,6 +350,38 @@ _establish_connection(struct https_request * const req,
 #endif /* HAVE_GETADDRINFO */
 }
 
+/* Provide implementations for HMAC_CTX_new and HMAC_CTX_free when
+ * building for OpenSSL versions older than 1.1.0
+ */
+#if OPENSSL_VERSION_NUMBER < 0x10100000L
+static void EVP_MD_CTX_free(EVP_MD_CTX *ctx)
+{
+    if (ctx != NULL) {
+        EVP_MD_CTX_cleanup(ctx);
```

Comment:
You're right, that would be a leak. I removed it during testing because calling free on that structure was crashing for some reason. I didn't get the chance to investigate further.